version: '3'

services: 
  keycloak:
    image:  127.0.0.1:5000/keycloak
    deploy:
      mode: replicated
      replicas: 1
      labels:
        - "traefik.enable=true"
        - "traefik.docker.network=traefik_public"
        - "traefik.http.routers.keycloak.entrypoints=websecure"
        - "traefik.http.routers.keycloak.rule=Host(`auth.localdomain`)"
        - "traefik.http.routers.keycloak.service=keycloak_service"
        - "traefik.http.routers.keycloak.tls=true"
        - "traefik.http.routers.keycloak.middlewares=keycloak_proto"
        - "traefik.http.services.keycloak_service.loadbalancer.server.port=8080"
        - "traefik.http.middlewares.keycloak_proto.headers.customrequestheaders.X-Forwarded-Proto=https"
    restart: on-failure
    environment:
      KC_PROXY_ADDRESS_FORWARDING: "true"
      KC_HOSTNAME_STRICT: "false"
      KC_PROXY: edge
      KC_HTTP_ENABLED: "true"
      KEYCLOAK_ADMIN: '{{ KEYCLOAK_ADMIN }}'
      KEYCLOAK_ADMIN_PASSWORD: '{{ KEYCLOAK_ADMIN_PASSWORD }}'
      KC_PROXY: edge
      KC_DB: postgres
      KC_DB_URL: jdbc:postgresql://db/keycloak
      KC_DB_PASSWORD: '{{ KC_DB_PASSWORD }}'
      KC_DB_USERNAME: '{{ KC_DB_USERNAME }}'
    volumes:
      - /opt/keycloak/realm.json:/opt/keycloak/data/import/realm.json
      - /data/logs/keycloak/keycloak.log:/logs/keycloak.log
    networks:
      traefik_public:
    depends_on:
      - postgres_db
    


networks:
  traefik_public:
    external: true    
    name: traefik_public
    